panther_docking/include/panther_docking/panther_charging_dock.hpp

```diff
@@ -199,45 +194,38 @@ class PantherChargingDock : public opennav_docking_core::ChargingDock
    */
   void updateStagingPoseAndPublish(const std::string & frame);
 
-  std::string name_;
   std::string base_frame_name_;
   std::string dock_frame_;
-  double panther_version_;
 
   rclcpp::Logger logger_{rclcpp::get_logger("PantherChargingDock")};
   rclcpp::Clock steady_clock_{RCL_STEADY_TIME};
+
   rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
```

Comment:
We may have cyclic dependency here with Node passed through the constructor. Once a `shared_ptr` object is created the ownership of an object is established. This may result in undefined behavior during destruction. A safer option is to create a class member type `std::weakt_ptr` and lock it in local scope once the node is needed.

Reply:
Frankly speaking, the node is locked in whole space of a charging dock plugin.  It is here
https://github.com/husarion/panther_ros/blob/ros2-docking/panther_docking/src/panther_charging_dock.cpp#L39

Do you prefer to check if we can lock the node everywhere we need it?
In `defineParameters`, `declareParameters` `updateDockPoseAndPublish` and `updateStagingPoseAndPublish`?

Reply:
That's what I meant. If u lock the weak pointer and create a member of type `share_ptr`, you gain ownership of the allocated memory.

Reply:
Sure. I found example here https://github.com/ros-navigation/navigation2/blob/c38fefd45531af496099d129b9ad5b8f407161be/nav2_navfn_planner/src/navfn_planner.cpp.


Added locking in necessary functions and tested configure function.